packages/vsx-registry/src/browser/vsx-extensions-contribution.ts

```diff
@@ -48,7 +48,8 @@ export class VSXExtensionsContribution extends AbstractViewContribution<VSXExten
     @inject(LabelProvider) protected readonly labelProvider: LabelProvider;
     @inject(ClipboardService) protected readonly clipboardService: ClipboardService;
     @inject(PreferenceService) protected readonly preferenceService: PreferenceService;
-    @inject(OVSXClientProvider) protected readonly clientProvider: OVSXClientProvider;
+    @inject(OVSXClientProvider) protected clientProvider: OVSXClientProvider;
```

Comment:
I somewhat accidentally removed the `readonly` modifier as I was dealing with these fields.

FWIW I think it's a mistake to make injected fields readonly. I don't see how it prevents any kind of issue. It actually prevents manual field injection when testing as we cannot assign values to the `readonly` fields.

I'll remove the `readonly` modifiers on injected fields in this class to make it more consistent then.

Reply:
As an old server-side developer I would prefer to have `readonly` everywhere it could be placed :) I would propose to do vice versa, to add `readonly` and configure the tests via tiny Inversify snippets. What do you think about it?

Reply:
To add to my case, consider the following [snippet](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgLIE8CSBbbBXMOAIwBsIBlaAN2CWQG8AoZF5KCOAEwHsQT1kMYBBKcAggC5k3IgCsICMM1bsuvfoOGiAQlJnzFylgGdu2CKghgAFt04AKAJRSq3YJyPSb0SzbtOXNw8AX0ZGTgUSOHZkBF5jMGRjaloIKQwcfEJSChSkMOSoGiQAOiERcWQAXgZg5AB6euRoKG4oZDhjZoAPAAcFSA9C4ogS03NfW05q5CdqgD5ahqaAFXJkYC6hEBRraAgAQkYgA):

```ts
interface MyImmutableService {
    readonly fieldA: object
    readonly fieldB: object
    someMethod(): void
    otherMethod(): void
}

declare const service: MyImmutableService

service.fieldA = {} // error as expected
service.someMethod = () => {} // TS is fine here!
```

If you want the methods to be readonly too you need to write things like:

```ts
interface MyImmutableService {
    readonly fieldA: object
    readonly fieldB: object
    readonly someMethod: () => void
    readonly otherMethod: () => void
}
```

But at this point you really denature service interface definition...

Long story short: I don't think we should care about making `readonly` fields in services while it may still make sense for "data-objects" that may be returned and passed around by APIs.

Reply:
I wrote about server-side for exact reasons you mentioned above: more immutability is better. If something could be made immutable, or at least declared immutable, I consider it is good. `readonly`, as far as I understand, should prevent somebody from assigning a value to a field accidentally in the code. And also it says to the reader "this field is not supposed to be changed after it is assigned". But please take all my arguments with a grain of salt, my Theia and TS experience should be considered as basic at this stage.
And just as a side note, for me the ability to create an instance of interface and start assigning its fields is quite crazy :) I understand why it works like this, but still...
As for `readonly` for DI fields, I would propose to make it a bit wider topic than a particular review. If there are any coding guidelines for Theia, what do they say about it? If nothing, I would propose to decide in a group (next meeting, maybe?) and then make it a guideline. And then write new code and rework old one, when there is an opportunity, according to the guideline. What do you think about it?